RusTaxi/Classes/Screens/VC/SlideShow Controller/MainController.swift

```diff
@@ -0,0 +1,242 @@
+		annotation.title = "1 мин."
+		mapView.addAnnotation(annotation)
+		locationManager.stopUpdatingLocation()
+		
+		print(location.altitude)
+		print(location.speed)
+	}
+	
+	func mapView(_ mapView: MKMapView, didUpdate userLocation: MKUserLocation) {
+		mapView.centerCoordinate = userLocation.location!.coordinate
+	}
+	
+	func mapView(_ mapView: MKMapView, viewFor annotation: MKAnnotation) -> MKAnnotationView? {
+		let annotationIdentifier = "MyCustomAnnotation"
+		guard !annotation.isKind(of: MKUserLocation.self) else {
+			return nil
+		}
+		var annotationView = mapView.dequeueReusableAnnotationView(withIdentifier: annotationIdentifier)
+		if annotationView == nil {
+			annotationView = CustomAnnotationView(annotation: annotation, reuseIdentifier: annotationIdentifier)
+			if case let annotationView as CustomAnnotationView = annotationView {
+				annotationView.isEnabled = true
+				annotationView.canShowCallout = false
+				annotationView.label = UILabel(frame: CGRect(x: -5.5, y: 11.0, width: 22.0, height: 16.5))
+				annotationView.tintColor = TaxiColor.orange
+				if let label = annotationView.label {
+					label.font = UIFont(name: "HelveticaNeue", size: 16.0)
+					label.textAlignment = .center
+					label.textColor = TaxiColor.black
+					label.adjustsFontSizeToFitWidth = true
+					annotationView.addSubview(label)
+				}
+			}
+		}
+		
+		if case let annotationView as CustomAnnotationView = annotationView {
+			annotationView.annotation = annotation
+			if let title = annotation.title,
+				let label = annotationView.label {
+				label.text = title
+			}
+		}
+		return annotationView
+	}
+	
+	func mapView(_ mapView: MKMapView, regionDidChangeAnimated animated: Bool) {
+		var center : CLLocationCoordinate2D = CLLocationCoordinate2D()
+		let mapLatitude = mapView.centerCoordinate.latitude
+		let mapLongtitude = mapView.centerCoordinate.longitude
+		let ceo: CLGeocoder = CLGeocoder()
+		center.latitude = mapLatitude
+		center.longitude = mapLongtitude
+		let loc: CLLocation = CLLocation(latitude:center.latitude, longitude: center.longitude)
+		ceo.reverseGeocodeLocation(loc, completionHandler:
+			{(placemarks, error) in
+				if (error != nil)
+				{
+					print("reverse geodcode fail: \(error!.localizedDescription)")
+				}
+				let pm = placemarks! as [CLPlacemark]
+				if pm.count > 0 {
+					let pm = placemarks![0]
+					var addressString : String = ""
+					var countryString: String = ""
+					if pm.thoroughfare != nil {
+						addressString = addressString + pm.thoroughfare! + ", "
+					}
+					
+					if pm.subThoroughfare != nil {
+						addressString = addressString + pm.subThoroughfare!
+					}
+					
+					if pm.locality != nil {
+						countryString = countryString + pm.locality! + ", "
+					}
+
+					if pm.country != nil {
+						countryString = countryString + pm.country! + " "
+					}
+					
+					self.addressModels[0].address = addressString
```

Comment:
are you sure, that zero index is exist in this array? unsafe.
you can use `array.contains(index: <Int>)` function in Sinder